// baseui/map-marker/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  FixedMarker: true,
  FloatingMarker: true,
  LocationPuck: true,
  FloatingRouteMarker: true,
  FLOATING_MARKER_ANCHOR_POSITIONS: true,
  FLOATING_MARKER_SIZES: true,
  FLOATING_MARKER_ANCHOR_TYPES: true,
  NEEDLE_SIZES: true,
  PINHEAD_SIZES_SHAPES: true,
  BADGE_ENHANCER_SIZES: true,
  LABEL_ENHANCER_POSITIONS: true,
  KIND: true,
  PINHEAD_DIMENSIONS: true,
  LOCATION_PUCK_SIZES: true,
  LOCATION_PUCK_TYPES: true,
  FLOATING_ROUTE_MARKER_ANCHOR_POSITIONS: true,
  calculateFloatingRouteMarkerOffsets: true
};
Object.defineProperty(exports, "BADGE_ENHANCER_SIZES", {
  enumerable: true,
  get: function () {
    return _constants.BADGE_ENHANCER_SIZES;
  }
});
Object.defineProperty(exports, "FLOATING_MARKER_ANCHOR_POSITIONS", {
  enumerable: true,
  get: function () {
    return _constants.FLOATING_MARKER_ANCHOR_POSITIONS;
  }
});
Object.defineProperty(exports, "FLOATING_MARKER_ANCHOR_TYPES", {
  enumerable: true,
  get: function () {
    return _constants.FLOATING_MARKER_ANCHOR_TYPES;
  }
});
Object.defineProperty(exports, "FLOATING_MARKER_SIZES", {
  enumerable: true,
  get: function () {
    return _constants.FLOATING_MARKER_SIZES;
  }
});
Object.defineProperty(exports, "FLOATING_ROUTE_MARKER_ANCHOR_POSITIONS", {
  enumerable: true,
  get: function () {
    return _constants.FLOATING_ROUTE_MARKER_ANCHOR_POSITIONS;
  }
});
Object.defineProperty(exports, "FixedMarker", {
  enumerable: true,
  get: function () {
    return _fixedMarker.default;
  }
});
Object.defineProperty(exports, "FloatingMarker", {
  enumerable: true,
  get: function () {
    return _floatingMarker.default;
  }
});
Object.defineProperty(exports, "FloatingRouteMarker", {
  enumerable: true,
  get: function () {
    return _floatingRouteMarker.default;
  }
});
Object.defineProperty(exports, "KIND", {
  enumerable: true,
  get: function () {
    return _constants.KIND;
  }
});
Object.defineProperty(exports, "LABEL_ENHANCER_POSITIONS", {
  enumerable: true,
  get: function () {
    return _constants.LABEL_ENHANCER_POSITIONS;
  }
});
Object.defineProperty(exports, "LOCATION_PUCK_SIZES", {
  enumerable: true,
  get: function () {
    return _constants.LOCATION_PUCK_SIZES;
  }
});
Object.defineProperty(exports, "LOCATION_PUCK_TYPES", {
  enumerable: true,
  get: function () {
    return _constants.LOCATION_PUCK_TYPES;
  }
});
Object.defineProperty(exports, "LocationPuck", {
  enumerable: true,
  get: function () {
    return _locationPuck.default;
  }
});
Object.defineProperty(exports, "NEEDLE_SIZES", {
  enumerable: true,
  get: function () {
    return _constants.NEEDLE_SIZES;
  }
});
Object.defineProperty(exports, "PINHEAD_DIMENSIONS", {
  enumerable: true,
  get: function () {
    return _constants.PINHEAD_DIMENSIONS;
  }
});
Object.defineProperty(exports, "PINHEAD_SIZES_SHAPES", {
  enumerable: true,
  get: function () {
    return _constants.PINHEAD_SIZES_SHAPES;
  }
});
Object.defineProperty(exports, "calculateFloatingRouteMarkerOffsets", {
  enumerable: true,
  get: function () {
    return _calculateOffsets.calculateFloatingRouteMarkerOffsets;
  }
});
var _fixedMarker = _interopRequireDefault(require("./fixed-marker"));
var _floatingMarker = _interopRequireDefault(require("./floating-marker"));
var _locationPuck = _interopRequireDefault(require("./location-puck"));
var _floatingRouteMarker = _interopRequireDefault(require("./floating-route-marker"));
var _constants = require("./constants");
var _calculateOffsets = require("./calculate-offsets");
var _types = require("./types");
Object.keys(_types).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _types[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _types[key];
    }
  });
});
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
