// yoga/dist/index.js
"use strict";
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var ApolloServer = __importStar(require("apollo-server-express"));
exports.ApolloServer = ApolloServer;
var express_1 = __importDefault(require("express"));
exports.express = express_1.default;
var config_1 = require("./config");
__export(require("nexus"));
__export(require("nexus-prisma"));
config_1.injectCustomEnvironmentVariables();
function yogaConfig(opts) {
    return opts;
}
exports.yogaConfig = yogaConfig;
function yogaEject(opts) {
    return opts;
}
exports.yogaEject = yogaEject;
function yogaExpress(fn) {
    return fn;
}
exports.yogaExpress = yogaExpress;
function yogaContext(ctx) {
    return ctx;
}
exports.yogaContext = yogaContext;
