// yoga/dist/nexusDefaults.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
var path_1 = require("path");
// TODO: shouldGenerateArtifacts should use process.env.NODE_ENV of the app using yoga, and not yoga process itself
function makeSchemaDefaults(config, types, prismaClientDir) {
    return {
        types: types,
        outputs: {
            schema: config.output.schemaPath,
            typegen: config.output.typegenPath,
        },
        nonNullDefaults: {
            input: true,
            output: true,
        },
        // @ts-ignore
        typegenAutoConfig: __assign({ sources: (config.contextPath
                ? [
                    {
                        source: config.contextPath,
                        alias: 'ctx',
                    },
                ]
                : []).concat((prismaClientDir
                ? [
                    {
                        source: path_1.join(prismaClientDir, 'index.ts'),
                        alias: 'prisma',
                    },
                ]
                : []), (config.typesPath
                ? [
                    {
                        source: config.typesPath,
                        alias: 'types',
                    },
                ]
                : [])) }, (config.contextPath ? { contextType: 'ctx.Context' } : {})),
    };
}
exports.makeSchemaDefaults = makeSchemaDefaults;
