// baseui/datepicker/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WEEKDAYS = exports.STATE_CHANGE_TYPE = exports.RANGED_CALENDAR_BEHAVIOR = exports.ORIENTATION = exports.ISO_MONTH_FORMAT = exports.ISO_FORMAT = exports.INPUT_ROLE = exports.DISPLAY_FORMAT = exports.DENSITY = exports.DEFAULT_MONTHS = void 0;
/*
Copyright (c) Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
const DISPLAY_FORMAT = exports.DISPLAY_FORMAT = 'L';
const ISO_FORMAT = exports.ISO_FORMAT = 'YYYY-MM-DD';
const ISO_MONTH_FORMAT = exports.ISO_MONTH_FORMAT = 'YYYY-MM';
const ORIENTATION = exports.ORIENTATION = Object.freeze({
  horizontal: 'horizontal',
  vertical: 'vertical'
});
const STATE_CHANGE_TYPE = exports.STATE_CHANGE_TYPE = Object.freeze({
  change: 'change',
  moveUp: 'moveUp',
  moveDown: 'moveDown',
  moveLeft: 'moveLeft',
  moveRight: 'moveRight',
  mouseOver: 'mouseOver',
  mouseLeave: 'mouseLeave'
});
const WEEKDAYS = exports.WEEKDAYS = [0, 1, 2, 3, 4, 5, 6];
const DEFAULT_MONTHS = exports.DEFAULT_MONTHS = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11];
const DENSITY = exports.DENSITY = {
  high: 'high',
  default: 'default'
};
const INPUT_ROLE = exports.INPUT_ROLE = {
  startDate: 'startDate',
  endDate: 'endDate'
};
const RANGED_CALENDAR_BEHAVIOR = exports.RANGED_CALENDAR_BEHAVIOR = {
  default: 'default',
  locked: 'locked'
};
