// baseui/app-nav-bar/user-profile-tile.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = UserProfileTile;
var React = _interopRequireWildcard(require("react"));
var _avatar = require("../avatar");
var _overrides = require("../helpers/overrides");
var _typography = require("../typography");
var _styledComponents = require("./styled-components");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
/*
Copyright (c) Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/

function UserProfileTile(props) {
  const {
    overrides = {},
    username,
    usernameSubtitle,
    userImgUrl
  } = props;
  const [UserProfileTileContainer, userProfileTileContainerProps] = (0, _overrides.getOverrides)(overrides.UserProfileTileContainer, _styledComponents.StyledUserProfileTileContainer);
  const [UserProfilePictureContainer, userProfilePictureContainerProps] = (0, _overrides.getOverrides)(overrides.UserProfilePictureContainer, _styledComponents.StyledUserProfilePictureContainer);
  const [UserProfileInfoContainer, userProfileInfoContainerProps] = (0, _overrides.getOverrides)(overrides.UserProfileInfoContainer, _styledComponents.StyledUserProfileInfoContainer);
  return (
    /*#__PURE__*/
    // Replace with a  profile tile renderer: renderUserProfileTile()
    React.createElement(UserProfileTileContainer, userProfileTileContainerProps, /*#__PURE__*/React.createElement(UserProfilePictureContainer, userProfilePictureContainerProps, /*#__PURE__*/React.createElement(_avatar.Avatar, {
      name: username || '',
      src: userImgUrl,
      size: '48px'
    })), /*#__PURE__*/React.createElement(UserProfileInfoContainer, userProfileInfoContainerProps, /*#__PURE__*/React.createElement(_typography.LabelMedium, null, username), usernameSubtitle ? /*#__PURE__*/React.createElement(_typography.ParagraphSmall, {
      marginTop: "0",
      marginBottom: "0"
    }, usernameSubtitle) : null))
  );
}
