// baseui/styles/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  ThemeConsumer: true,
  ThemeProvider: true,
  createThemedStyled: true,
  createThemedWithStyle: true,
  createThemedUseStyletron: true,
  styled: true,
  withStyle: true,
  useStyletron: true,
  withWrapper: true,
  hexToRgb: true,
  expandBorderStyles: true
};
exports.ThemeConsumer = void 0;
Object.defineProperty(exports, "ThemeProvider", {
  enumerable: true,
  get: function () {
    return _themeProvider.default;
  }
});
Object.defineProperty(exports, "createThemedStyled", {
  enumerable: true,
  get: function () {
    return _styled.createThemedStyled;
  }
});
Object.defineProperty(exports, "createThemedUseStyletron", {
  enumerable: true,
  get: function () {
    return _styled.createThemedUseStyletron;
  }
});
Object.defineProperty(exports, "createThemedWithStyle", {
  enumerable: true,
  get: function () {
    return _styled.createThemedWithStyle;
  }
});
Object.defineProperty(exports, "expandBorderStyles", {
  enumerable: true,
  get: function () {
    return _util.expandBorderStyles;
  }
});
Object.defineProperty(exports, "hexToRgb", {
  enumerable: true,
  get: function () {
    return _util.hexToRgb;
  }
});
Object.defineProperty(exports, "styled", {
  enumerable: true,
  get: function () {
    return _styled.styled;
  }
});
Object.defineProperty(exports, "useStyletron", {
  enumerable: true,
  get: function () {
    return _styled.useStyletron;
  }
});
Object.defineProperty(exports, "withStyle", {
  enumerable: true,
  get: function () {
    return _styled.withStyle;
  }
});
Object.defineProperty(exports, "withWrapper", {
  enumerable: true,
  get: function () {
    return _styled.withWrapper;
  }
});
var _themeProvider = _interopRequireWildcard(require("./theme-provider"));
var _styled = require("./styled");
var _util = require("./util");
var _types = require("./types");
Object.keys(_types).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _types[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _types[key];
    }
  });
});
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
/*
Copyright (c) Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/

const ThemeConsumer = exports.ThemeConsumer = _themeProvider.ThemeContext.Consumer;
