// baseui/toast/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  toaster: true,
  ToasterContainer: true,
  Toast: true,
  KIND: true,
  PLACEMENT: true,
  TYPE: true,
  StyledRoot: true,
  StyledBody: true,
  StyledCloseIcon: true
};
Object.defineProperty(exports, "KIND", {
  enumerable: true,
  get: function () {
    return _constants.KIND;
  }
});
Object.defineProperty(exports, "PLACEMENT", {
  enumerable: true,
  get: function () {
    return _constants.PLACEMENT;
  }
});
Object.defineProperty(exports, "StyledBody", {
  enumerable: true,
  get: function () {
    return _styledComponents.Body;
  }
});
Object.defineProperty(exports, "StyledCloseIcon", {
  enumerable: true,
  get: function () {
    return _styledComponents.CloseIconSvg;
  }
});
Object.defineProperty(exports, "StyledRoot", {
  enumerable: true,
  get: function () {
    return _styledComponents.Root;
  }
});
Object.defineProperty(exports, "TYPE", {
  enumerable: true,
  get: function () {
    return _constants.TYPE;
  }
});
Object.defineProperty(exports, "Toast", {
  enumerable: true,
  get: function () {
    return _toast.default;
  }
});
Object.defineProperty(exports, "ToasterContainer", {
  enumerable: true,
  get: function () {
    return _toaster.ToasterContainer;
  }
});
Object.defineProperty(exports, "toaster", {
  enumerable: true,
  get: function () {
    return _toaster.default;
  }
});
var _toaster = _interopRequireWildcard(require("./toaster"));
var _toast = _interopRequireDefault(require("./toast"));
var _constants = require("./constants");
var _styledComponents = require("./styled-components");
var _types = require("./types");
Object.keys(_types).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _types[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _types[key];
    }
  });
});
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
