// baseui/themes/dark-theme/create-dark-theme.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = createDarkTheme;
var _animation = _interopRequireDefault(require("../shared/animation"));
var _borders = _interopRequireDefault(require("./borders"));
var _breakpoints = _interopRequireDefault(require("../shared/breakpoints"));
var _deepMerge = _interopRequireDefault(require("../../utils/deep-merge"));
var _utils = require("../utils");
var _colorTokens = _interopRequireDefault(require("./color-tokens"));
var _tokens = require("../../tokens");
var _colorComponentTokens = _interopRequireDefault(require("./color-component-tokens"));
var _colorSemanticTokens = _interopRequireDefault(require("./color-semantic-tokens"));
var _typography = _interopRequireDefault(require("../shared/typography"));
var _grid = _interopRequireDefault(require("../shared/grid"));
var _lighting = _interopRequireDefault(require("../shared/lighting"));
var _mediaQuery = _interopRequireDefault(require("../shared/media-query"));
var _sizing = _interopRequireDefault(require("../shared/sizing"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
/*
Copyright (c) Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/

function createDarkTheme(overrides) {
  const foundationColorTokens = {
    ..._colorTokens.default,
    ...(0, _utils.getFoundationColorTokenOverrides)(overrides?.colors)
  };
  const semanticColorTokens = (0, _colorSemanticTokens.default)(foundationColorTokens);
  const componentColorTokens = (0, _colorComponentTokens.default)(foundationColorTokens);
  const theme = {
    animation: _animation.default,
    borders: _borders.default,
    breakpoints: _breakpoints.default,
    colors: {
      ..._tokens.colors,
      ...foundationColorTokens,
      ...semanticColorTokens,
      ...componentColorTokens
    },
    direction: 'auto',
    grid: _grid.default,
    lighting: _lighting.default,
    mediaQuery: _mediaQuery.default,
    sizing: _sizing.default,
    typography: _typography.default,
    // TODO(#2318) Remove in v11, the next major version.
    // Do not use.
    zIndex: {
      modal: 2000
    }
  };
  return (0, _deepMerge.default)(theme, overrides);
}
