// create-yoga/dist/templates.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const chalk_1 = require("chalk");
exports.defaultTemplate = {
    name: 'minimal-yoga',
    description: 'Basic starter template ',
    repo: {
        uri: 'https://github.com/prisma/yoga2',
        branch: 'master',
        path: '/examples/minimal',
    },
    postIntallMessage: `
Your template has been successfully set up!
  
Here are the next steps to get you started:
  1. Run ${chalk_1.default.yellow(`yarn dev`)} (Starts the GraphQL server)
  2. That's it !
  `,
};
exports.availableTemplates = [
    exports.defaultTemplate,
    {
        name: 'db-yoga',
        description: 'Template with Prisma database support',
        repo: {
            uri: 'https://github.com/prisma/yoga2',
            branch: 'master',
            path: '/examples/with-db',
        },
        postIntallMessage: `
Your template has been successfully set up!
  
Here are the next steps to get you started:
  1. Run ${chalk_1.default.yellow(`yarn prisma deploy`)} (choose a Demo server for a quicker startup)
  2. Run ${chalk_1.default.yellow(`yarn dev`)} (Starts the GraphQL server)
  3. That's it !
  `,
    },
];
exports.templatesNames = exports.availableTemplates
    .map(t => `\`${t.name}\``)
    .join(', ');
