// baseui/helper/helper-steps.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.HelperSteps = HelperSteps;
var React = _interopRequireWildcard(require("react"));
var _arrowLeft = _interopRequireDefault(require("../icon/arrow-left"));
var _arrowRight = _interopRequireDefault(require("../icon/arrow-right"));
var _check = _interopRequireDefault(require("../icon/check"));
var _button = require("../button");
var _styles = require("../styles");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
/*
Copyright (c) Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/

function HelperSteps({
  index,
  length,
  onFinish,
  onPrev,
  onNext
}) {
  const [css, theme] = (0, _styles.useStyletron)();
  const isLast = index === length - 1;
  return /*#__PURE__*/React.createElement("div", {
    className: css({
      alignItems: 'center',
      display: 'flex',
      justifyContent: 'space-between'
    })
  }, /*#__PURE__*/React.createElement(_button.Button, {
    disabled: index === 0,
    kind: _button.KIND.secondary,
    onClick: onPrev,
    shape: _button.SHAPE.circle
  }, /*#__PURE__*/React.createElement(_arrowLeft.default, {
    size: 20
  })), /*#__PURE__*/React.createElement("div", {
    className: css({
      display: 'flex'
    })
  }, new Array(length).fill(undefined).map((_, i) => {
    return /*#__PURE__*/React.createElement("div", {
      key: i,
      className: css({
        height: '8px',
        width: '8px',
        backgroundColor: i === index ? theme.colors.contentPrimary : theme.colors.backgroundTertiary,
        borderTopLeftRadius: '50%',
        borderTopRightRadius: '50%',
        borderBottomRightRadius: '50%',
        borderBottomLeftRadius: '50%',
        ':not(:last-child)': {
          marginRight: '8px'
        }
      })
    });
  })), /*#__PURE__*/React.createElement(_button.Button, {
    kind: isLast ? _button.KIND.primary : _button.KIND.secondary,
    onClick: isLast ? onFinish : onNext,
    shape: _button.SHAPE.circle
  }, isLast ? /*#__PURE__*/React.createElement(_check.default, {
    size: 20
  }) : /*#__PURE__*/React.createElement(_arrowRight.default, {
    size: 20
  })));
}
