// baseui/progress-steps/numbered-step.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var React = _interopRequireWildcard(require("react"));
var _overrides = require("../helpers/overrides");
var _styledComponents = require("./styled-components");
var _check = _interopRequireDefault(require("../icon/check"));
var _constants = require("./constants");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); } /*
Copyright (c) Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
function NumberedStep({
  overrides = {},
  isCompleted,
  isCurrent,
  isActive,
  isRightBeforeActive,
  isLast = false,
  orientation = _constants.ORIENTATION.vertical,
  title,
  step,
  alwaysShowDescription,
  children
}) {
  const [Root, rootProps] = (0, _overrides.getOverrides)(overrides.Root, _styledComponents.StyledNumberStep);
  const [Icon, iconProps] = (0, _overrides.getOverrides)(overrides.Icon, _styledComponents.StyledNumberIcon);
  const [Tail, tailProps] = (0, _overrides.getOverrides)(overrides.Tail, _styledComponents.StyledNumberContentTail);
  const [Content, contentProps] = (0, _overrides.getOverrides)(overrides.Content, _styledComponents.StyledContent);
  const [Title, titleProps] = (0, _overrides.getOverrides)(overrides.Title, _styledComponents.StyledContentTitle);
  const [Description, descriptionProps] = (0, _overrides.getOverrides)(overrides.Description, _styledComponents.StyledContentDescription);
  const [CheckIcon, checkIconProps] = (0, _overrides.getOverrides)(overrides.Icon, _check.default);
  const sharedProps = {
    $isCompleted: isCompleted,
    $isCurrent: isCurrent,
    $isActive: isActive,
    $isRightBeforeActive: isRightBeforeActive,
    $orientation: orientation
  };
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Root, _extends({}, sharedProps, rootProps), /*#__PURE__*/React.createElement(Icon, _extends({}, sharedProps, iconProps), !isCompleted && /*#__PURE__*/React.createElement("span", null, step), isCompleted && /*#__PURE__*/React.createElement(CheckIcon, _extends({
    size: 30
  }, checkIconProps))), !isLast && orientation === _constants.ORIENTATION.vertical && /*#__PURE__*/React.createElement(Tail, _extends({}, sharedProps, tailProps)), /*#__PURE__*/React.createElement(Content, _extends({}, sharedProps, contentProps), /*#__PURE__*/React.createElement(Title, _extends({}, sharedProps, titleProps), title), /*#__PURE__*/React.createElement(Description, descriptionProps, (isActive || alwaysShowDescription) && children))), !isLast && orientation === _constants.ORIENTATION.horizontal && /*#__PURE__*/React.createElement(Tail, _extends({}, sharedProps, tailProps, {
    "aria-hidden": "true"
  })));
}
var _default = exports.default = NumberedStep;
