// baseui/breadcrumbs/styled-components.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StyledSeparator = exports.StyledRoot = exports.StyledListItem = exports.StyledList = void 0;
var _styles = require("../styles");
/*
Copyright (c) Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/

const StyledRoot = exports.StyledRoot = (0, _styles.styled)('nav', ({
  $theme
}) => {
  return {
    color: $theme.colors.breadcrumbsText,
    ...$theme.typography.font350
  };
});
StyledRoot.displayName = "StyledRoot";
StyledRoot.displayName = 'StyledRoot';
const StyledList = exports.StyledList = (0, _styles.styled)('ol', ({
  $theme
}) => {
  return {
    listStyleType: 'none',
    margin: 0,
    padding: 0,
    ...$theme.typography.font350
  };
});
StyledList.displayName = "StyledList";
StyledList.displayName = 'StyledList';
const StyledListItem = exports.StyledListItem = (0, _styles.styled)('li', ({
  $theme
}) => {
  return {
    display: 'inline-block',
    ...$theme.typography.font350
  };
});
StyledListItem.displayName = "StyledListItem";
StyledListItem.displayName = 'StyledListItem';
const StyledSeparator = exports.StyledSeparator = (0, _styles.styled)('div', ({
  $theme
}) => {
  return {
    display: 'inline-flex',
    alignItems: 'center',
    color: $theme.colors.breadcrumbsSeparatorFill,
    marginLeft: $theme.sizing.scale300,
    marginRight: $theme.sizing.scale300,
    verticalAlign: 'middle'
  };
});
StyledSeparator.displayName = "StyledSeparator";
StyledSeparator.displayName = 'StyledSeparator';
